File: rxbinding/src/main/java/com/forceson/rxbinding/widget/RxProgressBar.java
package com.forceson.rxbinding.widget;

import android.widget.ProgressBar;

import io.reactivex.functions.Consumer;

/**
 * Created by son on 2020-01-25.
 */
public class RxProgressBar {
    public static Consumer<? super Integer> incrementProgressBy(final ProgressBar view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                view.incrementProgressBy(value);
            }
        };
    }

    public static Consumer<? super Integer> incrementSecondaryProgressBy(final ProgressBar view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                view.incrementSecondaryProgressBy(value);
            }
        };
    }

    public static Consumer<? super Boolean> setIndeterminate(final ProgressBar view) {
        return new Consumer<Boolean>() {
            @Override
            public void accept(Boolean value) {
                view.setIndeterminate(value);
            }
        };
    }

    public static Consumer<? super Integer> setMax(final ProgressBar view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                view.setMax(value);
            }
        };
    }

    public static Consumer<? super Integer> setProgress(final ProgressBar view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                view.setProgress(value);
            }
        };
    }

    public static Consumer<? super Integer> setSecondaryProgress(final ProgressBar view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                view.setSecondaryProgress(value);
            }
        };
    }
}

File: rxbinding-kotlin/src/main/java/com/forceson/rxbinding/kotlin/widget/RxProgressBar.kt
package com.forceson.rxbinding.kotlin.widget

import android.widget.ProgressBar
import com.forceson.rxbinding.widget.RxProgressBar
import io.reactivex.functions.Consumer

/**
 * Created by son on 2020-02-10.
 */

fun ProgressBar.incrementProgress(): Consumer<in Int> = RxProgressBar.incrementProgressBy(this)

fun ProgressBar.incrementSecondaryProgress(): Consumer<in Int> = RxProgressBar.incrementSecondaryProgressBy(this)

fun ProgressBar.indeterminate(): Consumer<in Boolean> = RxProgressBar.setIndeterminate(this)

fun ProgressBar.max(): Consumer<in Int> = RxProgressBar.setMax(this)

fun ProgressBar.progress(): Consumer<in Int> = RxProgressBar.setProgress(this)

fun ProgressBar.secondaryProgress(): Consumer<in Int> = RxProgressBar.setSecondaryProgress(this)